src/QirRuntime/lib/QIR/bridge-qis.ll

```diff
@@ -281,3 +281,48 @@ define void @__quantum__qis__z__ctl(%Array* %.ctls, %Qubit* %.q) {
 }
 
 
+;===============================================================================
+; quantum.qis math functions
+;
+
+; LLVM intrinsics (https://llvm.org/docs/LangRef.html):
+declare double      @llvm.sqrt.f64(double %.val)
+;declare double      @llvm.ceil.f64(double %Val)
+
+; Native implementations:
+declare i1          @quantum__qis__isnan__body(double %d)
+declare double      @quantum__qis__infinity__body()
+declare i1          @quantum__qis__isinf__body(double %d)
+
+; API for the user code:
+define double @__quantum__qis__nan__body() {                ; http://www.cplusplus.com/reference/cmath/nan-function/
+  %result = call double @llvm.sqrt.f64(double -1.0)         ; sqrt(<negative>) -> NaN
+  ret double %result
+}
+
+define i1 @__quantum__qis__isnan__body(double %d) {         ; http://www.cplusplus.com/reference/cmath/isnan/
+  %result = call i1 @quantum__qis__isnan__body(double %d)
+  ret i1 %result
+}
+
+define double @__quantum__qis__infinity__body() {           ; https://en.cppreference.com/w/c/numeric/math/INFINITY
+  %result = call double @quantum__qis__infinity__body()
+  ret double %result
+}
+
+define i1 @__quantum__qis__isinf__body(double %d) {         ; https://en.cppreference.com/w/cpp/numeric/math/isinf
+  %result = call i1 @quantum__qis__isinf__body(double %d)   
+  ret i1 %result
+}
+
+define double @__quantum__qis__sqrt__body(double %d) {      ; https://en.cppreference.com/w/cpp/numeric/math/sqrt
+  %result = call double @llvm.sqrt.f64(double %d)           
+  ret double %result
+}
+
+;define i64 @__quantum__qis__ceiling__body(double %d) {

```

Comment:
Same comment here; can we drop this and if not, could we add a note and GH issue #?

Reply:
This is work in progress to be added in the next commit/PR.
(I was working on it when I got the first feedback about Sqrt, so I postponed Ceil until I resolve all for Sqrt)

Reply:
Ceiling shouldn't be needed at all post-#497, though? Ceiling, Floor, and Round are all expressed in terms of Truncate, such that that's the only intrinsic needed to support those.

Reply:
Thanks for letting me know on time, Chris. Removing Ceil from my plate, adding Truncate.